### PR TITLE
Add InvalidNetworkModal to site/pledge

### LINKED ISSUE #507: [pledge] add 'wrong network modal' to site
Good catch from @ShimonD-KlimaDAO we haven't migrated this modal yet. It should be easy to port over.

### PR DESCRIPTION
**Pull request by 0xAeterno**
## Description

Add modal to `/pledges` to prompt user to switch to the polygon network when they're on the wrong network.

#### Tech debt
The code is more a less a copy of the `InvalidNetworkModal` on app but uses the modal component created on site.
We should combine and create a generic modal component in lib that can be used across both sites. Issue raised here (todo)

## Related Ticket

Resolves #507 

## Changes

image

## Checklist

- [x] Building the site with `npm run build-site` works without errors
- [x] Building the app with `npm run build-app` works without errors
- [x] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

**Comment by Atmosfearful** [comment]
This will conflict with https://github.com/KlimaDAO/klimadao/pull/537, so i'll wait until we merge that

**Comment by 0xAeterno** [comment]
@Atmosfearful pr rebased onto staging

**Review by Atmosfearful — APPROVED** [review]
I'm not opposed to copypasta but if you want to make this a shared lib component, I don't see why not.
You just have to pass in the lingui string as prop, since `lib` doesnt have lingui

**Pull request merged**
